sdk/Pulumi.Tests/Provider/PropertyValueTests.cs

```diff
@@ -45,6 +45,33 @@ public async Task DeserializingBasicArgsWorks()
         Assert.Equal(10, basicArgs.PasswordLength);
     }
 
+    class SecretArgs : ResourceArgs
+    {
+        [Input("password", required: true)]
+        private Input<string>? _password;
+        public Input<string>? Password
+        {
+            get => _password;
+            set
+            {
+                var emptySecret = Output.CreateSecret(0);
+                _password = Output.Tuple<Input<string>?, int>(value, emptySecret).Apply(t => t.Item1);
+            }
+        }
+    }
+
+    [Fact]
+    public async Task DeserializingSecretArgsWorks()
+    {
+        var serializer = CreateSerializer();
+        var data = Object(
+            Pair("password", new PropertyValue(new PropertyValue("PW"))));
+
+        var basicArgs = await serializer.Deserialize<SecretArgs>(data);
+        var pw = await basicArgs.Password.ToOutput().GetValueAsync("");
+        Assert.Equal("PW", pw);
```

Comment:
Can this assert that Password is secret as well?

Reply:
done